src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -0,0 +1,621 @@
+                break;
+            case PaymentWorksConstants.PARTNERSHIP_TAX_CLASSIFICATION_INDICATOR : 
+                kfsOwnsershipTypeCode = PaymentWorksConstants.PaymentWorksTaxClassification.PARTNERSHIP.getTranslationToKfsOwnershipTypeCode();
+                break;
+            case PaymentWorksConstants.TRUST_ESTATE_TAX_CLASSIFICATION_INDICATOR : 
+                kfsOwnsershipTypeCode = PaymentWorksConstants.PaymentWorksTaxClassification.TRUST_ESTATE.getTranslationToKfsOwnershipTypeCode();
+                break;
+            case PaymentWorksConstants.LLC_TAXED_AS_C_CORPORATION_TAX_CLASSIFICATION_INDICATOR : 
+                kfsOwnsershipTypeCode = PaymentWorksConstants.PaymentWorksTaxClassification.LLC_TAXED_AS_C_CORPORATION.getTranslationToKfsOwnershipTypeCode();
+                break;
+            case PaymentWorksConstants.LLC_TAXED_AS_S_CORPORATION_TAX_CLASSIFICATION_INDICATOR : 
+                kfsOwnsershipTypeCode = PaymentWorksConstants.PaymentWorksTaxClassification.LLC_TAXED_AS_S_CORPORATION.getTranslationToKfsOwnershipTypeCode();
+                break;
+            case PaymentWorksConstants.LLC_TAXED_AS_PARTNERSHIP_TAX_CLASSIFICATION_INDICATOR: 
+                kfsOwnsershipTypeCode = PaymentWorksConstants.PaymentWorksTaxClassification.LLC_TAXED_AS_PARTNERSHIP.getTranslationToKfsOwnershipTypeCode();
+                break;
+            case PaymentWorksConstants.OTHER_TAX_CLASSIFICATION_INDICATOR: 
+                kfsOwnsershipTypeCode = PaymentWorksConstants.PaymentWorksTaxClassification.OTHER.getTranslationToKfsOwnershipTypeCode();
+                break;
+            default: break;
+        }
+        return kfsOwnsershipTypeCode;
+    }
+    
+    private KfsVendorDataWrapper populateBusinessLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
+        kfsVendorDataWrapper.getVendorDetail().setVendorName(pmwVendor.getRequestingCompanyLegalName());
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(false);
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(KFSConstants.EMPTY_STRING);
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(KFSConstants.EMPTY_STRING);
+        return kfsVendorDataWrapper;
+    }
+    
+    private KfsVendorDataWrapper populateFirstLastLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(true);
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(pmwVendor.getRequestingCompanyLegalFirstName());
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(pmwVendor.getRequestingCompanyLegalLastName());
+        return kfsVendorDataWrapper;
+    }
+
+    //CREATE SIMILAR ROUTINE FOR W8 WHEN FORIEGN VENDOR PROCESSING IS ADDED
+    private KfsVendorDataWrapper populateW9Attributes(KfsVendorDataWrapper kfsVendorDataWrapper, PaymentWorksVendor pmwVendor) {
```

Comment:
should this method be called populateW9AttributesForDomesticVendor? and instead of the comment maybe have a method pupulateW9AttributesForForeignVendor? Although it is probably best to have the current code only deal with the current implementation.

Reply:
Removed the comment.